File: Biblioteca-de-Trabajo/3. CODIGO/Prototipo v1.0/src/presentacion/VistaLogin.java
package presentacion;

import negocio.ServicioAutenticacion;
import negocio.GestorEventos;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class VistaLogin extends JFrame {
    private ServicioAutenticacion authSvc;
    private GestorEventos gestor;

    private JTextField txtCorreo;
    private JPasswordField txtPass;
    private JButton btnLogin;
    private JButton btnRecuperar;

    public VistaLogin(GestorEventos gestor) {
        this.gestor = gestor;
        this.authSvc = new ServicioAutenticacion(gestor);

        setTitle("FiCitas - Iniciar Sesión");
        setSize(350, 250);
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        setLocationRelativeTo(null);
        setLayout(new BorderLayout());

        JPanel panelCentro = new JPanel(new GridLayout(3, 2, 10, 10));
        panelCentro.setBorder(BorderFactory.createEmptyBorder(20, 20, 20, 20));

        panelCentro.add(new JLabel("Correo:"));
        txtCorreo = new JTextField();
        panelCentro.add(txtCorreo);

        panelCentro.add(new JLabel("Contraseña:"));
        txtPass = new JPasswordField();
        panelCentro.add(txtPass);

        btnLogin = new JButton("Ingresar");
        btnRecuperar = new JButton("Recuperar Contraseña");

        JPanel panelSur = new JPanel();
        panelSur.add(btnLogin);
        panelSur.add(btnRecuperar);

        add(new JLabel("Bienvenido a FiCitas", SwingConstants.CENTER), BorderLayout.NORTH);
        add(panelCentro, BorderLayout.CENTER);
        add(panelSur, BorderLayout.SOUTH);

        btnLogin.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                ingresar();
            }
        });

        btnRecuperar.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                new VistaRecuperarContrasena(authSvc).setVisible(true);
            }
        });

        // Focalizar cursor al abrir ventana
        addWindowListener(new java.awt.event.WindowAdapter() {
            public void windowOpened(java.awt.event.WindowEvent e) {
                txtCorreo.requestFocusInWindow();
            }
        });
    }

    private void ingresar() {
        String correo = txtCorreo.getText();
        String pass = new String(txtPass.getPassword());

        ServicioAutenticacion.AuthResult res = authSvc.iniciarSesion(correo, pass);

        if (res.exito) {
            JOptionPane.showMessageDialog(this, "Bienvenido " + res.usuario.getCorreo());
            
            if (res.perfil.getNombre().equals("Administrador")) {
                dispose();
                new VistaPerfiles(gestor, res.usuario, res.perfil).setVisible(true);
            } else if (res.perfil.getNombre().equals("Auxiliar")) {
                dispose();
                new VistaPacientes(gestor, res.usuario, res.perfil).setVisible(true);
            } else {
                JOptionPane.showMessageDialog(this, "Dashboard para " + res.perfil.getNombre() + " en construcción.");
                // Limpiar campos para otro intento
                txtPass.setText("");
                txtCorreo.requestFocusInWindow();
            }
        } else {
            JOptionPane.showMessageDialog(this, res.mensaje, "Error", JOptionPane.ERROR_MESSAGE);
        }
    }
}

File: Biblioteca-de-Trabajo/3. CODIGO/Prototipo v1.5/src/presentacion/VistaLogin.js
import { ServicioAutenticacion } from '../negocio/ServicioAutenticacion.js';

export class VistaLogin {
  constructor(gestor) {
    this.gestor = gestor;
    this.authSvc = new ServicioAutenticacion(gestor);
    this.render();
  }

  render() {
    const container = document.getElementById('app');
    container.innerHTML = `
      <div class="auth-screen">
        <div class="auth-card">
          <h1>FiCitas</h1>
          <p>Iniciar sesión</p>
          <form id="loginForm" class="auth-form">
            <label>Correo<input id="correo" type="email" required /></label>
            <label>Contraseña<input id="password" type="password" required /></label>
            <button class="primary-btn" type="submit">Ingresar</button>
          </form>
        </div>
      </div>
    `;

    document.getElementById('loginForm').addEventListener('submit', (e) => this.ingresar(e));
  }

  ingresar(e) {
    e.preventDefault();
    const correo = document.getElementById('correo').value;
    const pass = document.getElementById('password').value;
    const res = this.authSvc.iniciarSesion(correo, pass);

    if (res.exito) {
      alert(`Bienvenido ${res.usuario.correo}`);
      if (res.perfil.nombre === 'Administrador') {
        this.showVistaPerfiles();
      } else {
        this.showVistaPacientes();
      }
    } else {
      alert(res.mensaje);
    }
  }

  showVistaPerfiles() {
    import('./VistaPerfiles.js').then(({ VistaPerfiles }) => {
      new VistaPerfiles(this.gestor);
    });
  }

  showVistaPacientes() {
    import('./VistaPacientes.js').then(({ VistaPacientes }) => {
      new VistaPacientes(this.gestor);
    });
  }
}
